storage/operation/computation_result.go

```diff
@@ -27,24 +27,18 @@ func GetComputationResultUploadStatus(r storage.Reader, blockID flow.Identifier,
 // GetBlockIDsByStatus returns all IDs of stored ComputationResult instances.
 func GetBlockIDsByStatus(r storage.Reader, blockIDs *[]flow.Identifier,
 	targetUploadStatus bool) error {
-	return TraverseByPrefix(r, MakePrefix(codeComputationResults), func() (CheckFunc, CreateFunc, HandleFunc) {
-		var currKey flow.Identifier
-		check := func(key []byte) (bool, error) {
-			currKey = flow.HashToID(key[1:])
-			return true, nil
-		}
-
+	iterationFunc := func(keyCopy []byte, getValue func(destVal any) error) (bail bool, err error) {
 		var wasUploadCompleted bool
-		create := func() interface{} {
-			return &wasUploadCompleted
+		err = getValue(&wasUploadCompleted)
+		if err != nil {
+			return true, err
 		}
 
-		handle := func() error {
-			if blockIDs != nil && wasUploadCompleted == targetUploadStatus {
-				*blockIDs = append(*blockIDs, currKey)
-			}
-			return nil
+		if blockIDs != nil && wasUploadCompleted == targetUploadStatus {
```

Comment:
Condition `blockIDs != nil` isn't modified in this PR, but shouldn't `blockIDs == nil` be an error condition?

If `blockIDs == nil` and we can't append block IDs, what is the purpose of iterating blocks?